src/diffpy/utils/parsers/loadmetafile.py

```diff
@@ -0,0 +1,233 @@
+    existing = False
+    if pathlib.Path.is_file(dbname):
+        existing = True
+
+    # collect entry
+    with open(pdfname, 'r') as grfile:
+        data = {}
+
+        # add path
+        grpath = grfile.name
+        if show_path:
+            data.update({'PDF_path': grpath})
+
+        # add r, gr, and header metadata
+        data.update({'r': list(rv[:, 0]), 'gr': list(rv[:, 1])})
+        data.update(hddata)
+
+        # parse name using pathlib and generate json entry
+        name = pathlib.Path(grpath).name
+        entry = {name: data}
+
+    # check if supported type
+    extension = pathlib.Path(dbname).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {dbname} is not supported.")
+
+    # json
+    if extension == '.json':
+        # dump if non-existing
+        if not existing:
+            with open(dbname, 'w') as jsonfile:
+                pdfs = entry  # for return
+                json.dump(pdfs, jsonfile, indent=2)
+
+        # update if existing
+        else:
+            with open(dbname, 'r') as json_read:
+                pdfs = json.load(json_read)
+                pdfs.update(entry)
+            with open(dbname, 'w') as json_write:
+                # dump to string first for formatting
+                json.dump(pdfs, json_write, indent=2)
+
+    return pdfs
+
+
+def load_from_db(filename):
+    """Load a dictionary from a database file.
+
+    filename    -- database file to load from.
+
+    Returns a dictionary of database information.
+    """
+
+    # check if supported type
+    extension = pathlib.Path(filename).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {filename} is not supported.")
+
+    # json
+    if extension == '.json':
+        with open(filename, 'r') as json_file:
+            j_dict = json.load(json_file)
+
+    return j_dict
+
+
+def markup_PDF(hddata: dict, rv: list, muname=None):
+    # FIXME: may be better suited for REST API package, not diffpy.utils
+    """Put PDF file information into a dictionary.
+
+    hddata  -- Dictionary of metadata.
+    rv      -- List of (r, gr) pairs.
+    muname  -- file to save into (default None, no saving occurs).
+
+    Returns the dictionary loaded from/into markup file.
+    """
+
+    # gather data
+    data = {}
+    data.update({'r': list(rv[:, 0]), 'gr': list(rv[:, 1])})
+    data.update(hddata)
+
+    # return directly
+    if muname is None:
+        return data
+
+    # save to disk when enabled
+    extension = pathlib.Path(muname).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {muname} is not supported.")
+
+    # dumps into file, automatically overwrites
+    if extension == '.json':
+        with open(muname, 'w') as json_write:
+            json.dump(data, json_write, indent=2)
+
+    return data
+
+
+def markup_oneline(filename):
+    """Reformat lists in markup languages to take up only one line.
```

Comment:
I am not sure we need this function.  Making json readable by humans may not be worth it.  But it is ok to have it here I guess.

Reply:
Yea, probably not worth, can remove.